### PR TITLE
chore: optimize svg with svgo

### LINKED ISSUE #4222: [Bug] SVG id conflicts
### How to reproduce?

image

### Mask Version:

- [ ] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [ ] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [ ] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by UncleBill**
closes #4222

### CODE REVIEW

File: packages/icons/brands/Blocto.svg [hunk 1]
```diff
@@ -1,43 +1,21 @@
...
-                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
-                result="hardAlpha"
-            />
-            <feOffset dy="6" />
-            <feGaussianBlur stdDeviation="6" />
-            <feComposite in2="hardAlpha" operator="out" />
-            <feColorMatrix type="matrix" values="0 0 0 0 0.203922 0 0 0 0 0.521569 0 0 0 0 0.768627 0 0 0 0.2 0" />
-            <feBlend mode="normal" in2="effect1_backgroundBlur_270_89338" result="effect2_dropShadow_270_89338" />
-            <feBlend mode="normal" in="SourceGraphic" in2="effect2_dropShadow_270_89338" result="shape" />
+        <filter id="Blocto_svg__a" width="62" height="64" x="-9.5" y="-12" color-interpolation-filters="sRGB" filterUnits="userSpaceOnUse">
```
Comment (UncleBill) [inline packages/icons/brands/Blocto.svg:8]: `svgo` prefix the id, that how we resolve #4222

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "lint:ci-report": "pnpm run lint:ci --format=junit --output-file=reports/junit/eslint-results.xml --max-warnings=0",
     "postinstall": "gulp fix-lockfile",
     "prepare": "husky install",
+    "svgo": "svgo packages/icons/*/*.svg",
```
Comment (Jack-Works) [inline package.json:26]: please install it in packages/scripts or packages/icons.

File: packages/mask/.webpack/config.ts [hunk 3]
```diff
@@ -136,6 +137,20 @@ export function createConfiguration(rawFlags: BuildFlags): Configuration {
                         },
                     },
                 },
+                {
+                    test: /\.svg$/,
+                    use: [
+                        {
+                            loader: 'svgo-loader',
```
Comment (Jack-Works) [inline packages/mask/.webpack/config.ts]: please don't add loaders to convert non-js files into a js file.

and it seems like you doesn't use it, so you don't need to add this.

File: package.json (hunk 2 above)
Comment (UncleBill) [reply package.json:26]: No, need it to run in git hook. Installing in root directory would be better.

File: packages/mask/.webpack/config.ts (hunk 3 above)
Comment (UncleBill) [reply packages/mask/.webpack/config.ts]: We only refer svg with `new URL()`, in that way webpack won't convert it into JS but file.

It's to minify svg for release.

File: packages/mask/.webpack/config.ts [hunk 4]
```diff
@@ -136,6 +137,22 @@ export function createConfiguration(rawFlags: BuildFlags): Configuration {
                         },
                     },
                 },
+                mode === 'production'
+                    ? {
+                          test: /\.svg$/,
+                          use: [
+                              {
+                                  loader: 'svgo-loader',
+                                  // overrides
```
Comment (Jack-Works) [inline packages/mask/.webpack/config.ts]:
```suggestion
                                  loader: 'svgo-loader',
                                  dependency: (data) => {
                                      if (data === "") return false
                                      if (data !== 'url') throw new TypeError('The only import mode valid for a non-JS file is via new URL(). Current import mode: ' + data)
                                      return true
                                  },
                                  type: 'asset/resource'
```

File: package.json (hunk 2 above)
Comment (Jack-Works) [reply package.json:26]: I'm wondering, if you're running it in the source code, why you need to optimize it again in webpack?

Comment (UncleBill) [reply package.json:26]: Formatting in git hook, minifying in building.

Comment (UncleBill) [reply package.json:26]: The loader is not enabled in development mode.

Comment (Jack-Works) [reply package.json:26]: that's interesting

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**